web-ng/src/app/components/layer-dialog/layer-dialog.component.ts

```diff
@@ -134,7 +134,7 @@ export class LayerDialogComponent implements OnDestroy {
     const layer = new Layer(
       this.layerId,
       this.layer?.color || DEFAULT_LAYER_COLOR,
-      this.layer?.name?.set(this.lang, this.layerName),
+      StringMap({ [this.lang]: this.layerName }),
```

Comment:
This will only work for projects with only one language. In the future we'll support multiple languages, so `layerName` itself should be a map. Care to add a TODO for that?